lib/Parse/ParseDecl.cpp

```diff
@@ -8980,6 +8980,18 @@ Parser::parseDeclEnumCase(ParseDeclOptions Flags,
       if (ArgParams.isNull() || ArgParams.hasCodeCompletion())
         return ParserStatus(ArgParams);
     }
+    
+    // See if there are generic params.
+    auto genericParams = maybeParseGenericParams()
+      .getPtrOrNull();
+    if (genericParams) {
+      auto param = genericParams->getParams().front()->getStartLoc();
+      if (param) {
+        diagnose(param, diag::generic_param_cant_be_used_in_enum_case_decl);
+        Status.setIsParseError();
+        return Status;
```

Comment:
I don’t think we need to stop parsing here

Reply:
removed the `return`, but added a new validation to skip arguments following the generic declaration, this way if we have something like `case<T>(param: T)` we're complaining about the generics, but will not present `Consecutive declarations on a line must be separated by ';'`.

Also added a few more tests to cover this.